File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        int depth = 0;
        int current = 1;

        if (root == null) {
            return 0;
        }

        // while (!queue.isEmpty()) {
        //     TreeNode current = queue.poll();

        //     if (current.left != null) {
        //         queue.add(current.left);
        //     }

        //     if (current.right != null) {
        //         queue.add(current.right);
        //     }
        // }
    
        // depth = Math.max(current, depth)
        // if (root == null) [
        //     return 1;
        // ]
        // depth = Math.max(maxDepth(root.left), depth);
        // maxDepth(root.right);
        return dfs(root, depth);

    }

    public int dfs(TreeNode node, Integer current) {
        int depth = 0;
        if (node == null) {
            return current;
        }
        // dfs(node.left, current+1);
        depth = Math.max(dfs(node.left, current+1), depth);
        // dfs(node.right, current+1);
        depth = Math.max(dfs(node.right, current+1), depth);

        return depth;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.count = 0
        def dfs(r, max):
            if not r:
                return 0
            if r.val >= max:
                self.count += 1
                max = r.val
            
            dfs(r.left, max)
            dfs(r.right, max)
            # max = max(dfs(r.left), dfs(r.right))
        dfs(root, root.val)
        return self.count